app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java

```diff
@@ -218,10 +219,32 @@ public static boolean HandleGoodReading(String tagId, byte[] data1, final long C
         return HandleGoodReading(tagId, data1, CaptureDateTime, false, null, null);
     }
 
+
+    public static void SendLibrereading(final String tagId, byte[] data1, final long CaptureDateTime, byte []patchUid,  byte []patchInfo){
+        if(!Home.get_master()) {
+            return;
+        }
+        LibreBlock libreBlock = LibreBlock.getForTimestamp(CaptureDateTime);
+        if (libreBlock != null) {
+            // We already have this one, so we have already sent it, so let's not crate storms.
+            return;
+        }
+        // Create the object to send
+        libreBlock = LibreBlock.create(tagId, CaptureDateTime, data1, 0, patchUid, patchInfo);
+        if(libreBlock == null) {
+            Log.e(TAG, "Error could not create libreBlock for libre-allhouse");
+            return;
+        }
+        final String json = libreBlock.toExtendedJson();
+        
+        GcmActivity.pushLibreBlock(json);
+    
+    }
+    
     // returns true if checksum passed.
     public static boolean HandleGoodReading(final String tagId, byte[] data1, final long CaptureDateTime, final boolean allowUpload, byte []patchUid,  byte []patchInfo ) {
-
-
+        Log.e(TAG, "HandleGoodReading called");
+        SendLibrereading(tagId, data1, CaptureDateTime, patchUid, patchInfo);
```

Comment:
I am concerned that this change not being restricted to whole house coverage is an unrelated change to functionality which has a much wider scope and could increase the data load for followers somewhat but I understand why its desired so I think we will see how it goes.

I'm also concerned that this is the only code path where an incoming message can generate an outgoing message of the same time (as mentioned with the previous PR) because that has the potential for a loop. I can see that it is restricted with a database check but when phones get out of storage they can fail to write database records. I think for a loop to occur that would have to happen to multiple devices in the follower group which is less likely. 

It could be mitigated with an in memory check when sending the message that the message isn't the same one that was sent last time.